packages/block-library/src/heading/test/edit.native.js

```diff
@@ -0,0 +1,111 @@
+/**
+ * External dependencies
+ */
+import { shallow } from 'enzyme';
+
+/**
+ * WordPress dependencies
+ */
+import * as data from '@wordpress/data';
+import * as blockLibrary from '@wordpress/block-library';
+import * as blocks from '@wordpress/blocks';
+
+/**
+ * Internal dependencies
+ */
+import HeadingToolbar from '../heading-toolbar';
+import HeadingEdit from '../edit';
+
+const blockEditorSelect = data.select( 'core/block-editor' );
+const blockEditorDispatch = data.dispatch( 'core/block-editor' );
+
+const testAttributes = {
+	content: 'Heading block',
+	level: 4,
+	align: 'center',
+	style: { color: { text: '#003fa3' } },
+};
+
+describe( 'Heading', () => {
+	beforeAll( () => {
+		blockLibrary.registerCoreBlocks();
+		gutenbergSetup();
+		blockEditorDispatch.removeBlocks( blockEditorSelect.getBlocks() );
+	} );
+
+	it( 'Serializes all Heading attributes properly', () => {
+		const expectedHTML = `<!-- wp:heading {"align":"center","level":4,"style":{"color":{"text":"#003fa3"}}} -->
+<h4 class="has-text-align-center has-text-color" style="--wp--color--text:#003fa3">Heading block</h4>
+<!-- /wp:heading -->`;
+
+		insertHeadingBlockWithAttributes( testAttributes );
+
+		const finalHTML = blocks.serialize( blockEditorSelect.getBlocks() );
+
+		expect( finalHTML ).toEqual( expect.stringContaining( expectedHTML ) );
```

Comment:
This starts to look a lot like snapshot testing, right? The kind used already in Gutenberg to test the html output of blocks. See [this list test for example](https://github.com/WordPress/gutenberg/blob/6c95d81bba42681ee4e1f60ba4414bdde4167a3d/packages/e2e-tests/specs/editor/blocks/list.test.js#L29), where the editor contents are compared against [the serialized/saved block](https://github.com/WordPress/gutenberg/blob/6c95d81bba42681ee4e1f60ba4414bdde4167a3d/packages/e2e-tests/specs/editor/blocks/__snapshots__/list.test.js.snap#L71-L75).

A good thing with snapshot testing is that it's easier to maintain the "expectedHTML" strings since Jest itself provides the way to update them when needed. If you too think that this indeed looks like snapshot testing, WDYT about transforming this test into a snapshot test?

Reply:
True that. If we snapshot the serialised HTML, will be the same result without the need to "manually" write the HTML code in the test itself. Will give it a shot.